sstables/mx/partition_reversing_data_source.cc

```diff
@@ -0,0 +1,442 @@
+/*
+ * Copyright (C) 2021-present ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/iostream.hh>
+#include "partition_reversing_data_source.hh"
+#include "reader_permit.hh"
+#include "sstables/consumer.hh"
+#include "sstables/mx/bsearch_clustered_cursor.hh"
+#include "sstables/processing_result_generator.hh"
+#include "sstables/shared_sstable.hh"
+#include "sstables/sstables.hh"
+#include "sstables/types.hh"
+#include "vint-serialization.hh"
+
+namespace sstables {
+
+namespace mx {
+
+class partition_header_context : public data_consumer::continuous_data_consumer<partition_header_context> {
```

Comment:
Please add some words what this thing is.

Reply:
Beyond the general description, the mapping from the original sstable data file offsets to the reversed data stream should be explained (if it's really needed, I assume so because the index outputs offsets).

Reply:
Ah, the changelog says we don't do that. I'll read it carefully.